util/rlp/fuzz/fuzz_targets/untrusted.rs

```diff
@@ -0,0 +1,11 @@
+#![no_main]
+#![cfg(feature = "nightly")]
+#[macro_use] extern crate libfuzzer_sys;
+extern crate rlp;
+
+use rlp::UntrustedRlp;
+
+fuzz_target!(|data: &[u8]| {
+    let _ = UntrustedRlp::new(data);
```

Comment:
So the interesting thing about RLP is that actually, each item is either one of a few things:
  - A list of items, prefixed by its length
  - Special byte for empty data (treated as empty list)
  - A list of bytes, prefixed by its length

This is why it's called "recursive list prefix".
Take a look at [the spec](https://github.com/ethereum/wiki/wiki/RLP) for more details.
All of our strongly-typed decoding is implemented on top of those primitives.

What I'm suggesting is that we have the fuzzer try and generate data, which we treat as a bunch of nested lists of bytes (and try to walk accordingly). This will allow the fuzzer to attempt to explore almost all code paths in the decoder, possibly finding some well-formed RLP that leads to panic.